### DISCUSSION TITLE
How to create a disabled button?

### DISCUSSION THREAD

**Discussion by Virgil-N** [hecrj/iced#1728] [General]
How to create a disabled button? With disabled style.

**Comment by 13r0ck** [comment]
Just dont add the `on_press` method

**Reply by Virgil-N** [reply]
But how to write code if I want to create a normal button and change it's state to disabled when I pressed the button?

**Reply by 13r0ck** [reply]
Just an if and/or match statement from your application state. In the arms/blocks just return either a button or a button with an `on_press`. If you do that a lot, then you can create a helper function that does that for you

**Reply by Virgil-N** [reply]
> Just an if and/or match statement from your application state. In the arms/blocks just return either a button or a button with an `on_press`. If you do that a lot, then you can create a helper function that does that for you

Got it. Thanks.
